### PR TITLE
Update LSIF upload warnings.

### PR DESCRIPTION
**Pull request by efritz**
Remove some of the "coming soon to 3.10", rearranged data retention sections, and rephrased the warning that says not to upload more than 10-40 LSIF _dumps_ (which is _far_ too low). We've seen no impact on the dotcom instance running LSIF for two dozen repositories with ~300 active dumps (due to the default disk size).

### CODE REVIEW

File: doc/user/code_intelligence/lsif.md [hunk 1]
```diff
@@ -10,6 +10,8 @@
 
 Follow our [LSIF quickstart guide](lsif_quickstart.md) to manually generate and upload LSIF data for your repository. After you are satisfied with the result, you can upload LSIF data to a Sourcegraph instance using your existing [continuous integration infrastructure](lsif_in_ci.md), or using [GitHub Actions](lsif_on_github.md).
 
+LSIF support is still a relatively new feature. We recommend you start by uploading a smaller number of key repositories. Once you reach 50 to 100 repositories, we will be able to provide specific recommendations for ensuring stability and performance of your Sourcegraph instance. We are currently working on validating that the feature remains responsive with tens of thousands of repositories.
```
Comment (creachadair) [inline doc/user/code_intelligence/lsif.md]: I like this, but would suggest front-loading the key point of the message, which is to ensure the performance of the instance. This suggestion is just a rearrangement on that basis.

```suggestion
LSIF support is still a relatively new feature. We are currently working on validating that the feature remains responsive even with tens of thousands of repositories. To get started, we recommend you upload a smaller number of key repositories. Once you reach 50 to 100 repositories, we will be able to provide specific recommendations for ensuring stability and performance of your Sourcegraph instance.
```

File: doc/user/code_intelligence/lsif.md [hunk 2]
```diff
@@ -51,14 +49,9 @@ The following table gives a rough estimate for the space and time requirements f
...
+## Data retention policy
 
-
-
-
-
-## Cross-repository code intelligence
-
-Cross-repository code intelligence will only be powered by LSIF when **both** repositories have LSIF data. When the current file has LSIF data and the other repository doesn't, there will be no code intelligence results (we're working on fallback to fuzzy code intelligence for 3.10).
+The bulk of LSIF data is stored on-disk, and as code intelligence data for a commit ages it becomes less useful. Sourcegraph will automatically remove the least recently uploaded data if the amount of disk space falls above a threshold. This value can be changed via the `DBS_DIR_MAXIMUM_SIZE_BYTES` environment variable. The default value of this variable is `10737418240`, which is `1024 * 1024 * 1024 * 10` bytes, or `10` gigabytes.
```
Comment (creachadair) [inline doc/user/code_intelligence/lsif.md]:
```suggestion
The bulk of LSIF data is stored on-disk, and as code intelligence data for a commit ages it becomes less useful. Sourcegraph will automatically remove the least recently uploaded data if the amount of disk space falls below a configurable threshold. This value defaults to 10 GiB (10⨉2^30 = 10737418240  bytes), and can be changed via the `DBS_DIR_MAXIMUM_SIZE_BYTES` environment variable.
```

**Comment by efritz** [comment]
cc @slimsag for blocking approval

**Comment by efritz** [comment]
bump @slimsag

**Comment by efritz** [comment]
@nicksnyder suggested I merge this as it has enough approvals. If there is an issue with the wording or the general idea of relaxing the warning we can easily make an adjustment after merge.

**Pull request merged**